persistence/sql/persister_consent.go

```diff
@@ -392,41 +465,46 @@ func (p *Persister) listUserAuthenticatedClients(ctx context.Context, subject, s
 }
 
 func (p *Persister) CreateLogoutRequest(ctx context.Context, request *consent.LogoutRequest) error {
+	request.NID = p.NetworkID(ctx)
 	return errorsx.WithStack(p.Connection(ctx).Create(request))
 }
 
 func (p *Persister) AcceptLogoutRequest(ctx context.Context, challenge string) (*consent.LogoutRequest, error) {
-	if err := p.Connection(ctx).RawQuery("UPDATE hydra_oauth2_logout_request SET accepted=true, rejected=false WHERE challenge=?", challenge).Exec(); err != nil {
+	if err := p.Connection(ctx).RawQuery("UPDATE hydra_oauth2_logout_request SET accepted=true, rejected=false WHERE challenge=? AND nid = ?", challenge, p.NetworkID(ctx)).Exec(); err != nil {
 		return nil, sqlcon.HandleError(err)
 	}
 
 	return p.GetLogoutRequest(ctx, challenge)
 }
 
 func (p *Persister) RejectLogoutRequest(ctx context.Context, challenge string) error {
-	return errorsx.WithStack(
-		p.Connection(ctx).
-			RawQuery("UPDATE hydra_oauth2_logout_request SET rejected=true, accepted=false WHERE challenge=?", challenge).
-			Exec())
+	count, err := p.Connection(ctx).
+		RawQuery("UPDATE hydra_oauth2_logout_request SET rejected=true, accepted=false WHERE challenge=? AND nid = ?", challenge, p.NetworkID(ctx)).
+		ExecWithCount()
+	if count == 0 {
+		return errorsx.WithStack(x.ErrNotFound)
+	} else {
+		return errorsx.WithStack(err)
+	}
 }
 
 func (p *Persister) GetLogoutRequest(ctx context.Context, challenge string) (*consent.LogoutRequest, error) {
 	var lr consent.LogoutRequest
-	return &lr, sqlcon.HandleError(p.Connection(ctx).Where("challenge = ? AND rejected = FALSE", challenge).First(&lr))
+	return &lr, sqlcon.HandleError(p.Connection(ctx).Where("challenge = ? AND rejected = FALSE AND nid = ?", challenge, p.NetworkID(ctx)).First(&lr))
 }
 
 func (p *Persister) VerifyAndInvalidateLogoutRequest(ctx context.Context, verifier string) (*consent.LogoutRequest, error) {
 	var lr consent.LogoutRequest
 	return &lr, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		if err := c.Where("verifier=? AND was_used=FALSE AND accepted=TRUE AND rejected=FALSE", verifier).Select("challenge").First(&lr); err != nil {
+		if err := c.Where("verifier=? AND was_used=FALSE AND accepted=TRUE AND rejected=FALSE AND nid = ?", verifier, p.NetworkID(ctx)).Select("challenge").First(&lr); err != nil {
 			if err == sql.ErrNoRows {
 				return errorsx.WithStack(x.ErrNotFound)
 			}
 
 			return sqlcon.HandleError(err)
 		}
 
-		if err := c.RawQuery("UPDATE hydra_oauth2_logout_request SET was_used=TRUE WHERE verifier=?", verifier).Exec(); err != nil {
+		if err := c.RawQuery("UPDATE hydra_oauth2_logout_request SET was_used=TRUE WHERE verifier=? AND nid = ?", verifier, p.NetworkID(ctx)).Exec(); err != nil {
```

Comment:
QueryWith...

Reply:
The above comment was most likely made by mistake (RawQuery overrides pop's query builder)

Reply:
By the way, it's possible to reduce the number of database accesses from three to two by moving the WHERE check from the first SELECT into the UPDATE and querying the final result using verifier instead of challenge. Should I do it?

Note to self: the RawQuery UPDATE is necessary because we're updating by the verifier which isn't the primary key.

Reply:
On a second thought, I can see how this could be done without RawQuery in 3 DB roundtrips. So the options are
1. Refactor the function with 3 db roundtrips and no RawQuery
2. Refactor the function with 2 db roundtrips keeping the RawQuery

Reply:
2db roundtrips it is :)